root/edit/details/historic/EditReleaseAttributes.js

```diff
@@ -18,17 +18,29 @@ type EditReleaseAttributesEditT = {
     +changes: $ReadOnlyArray<{
       +releases: $ReadOnlyArray<ReleaseT | null>,
       +status: ReleaseStatusT | null,
-      +type: ReleaseGroupTypeT | null,
+      +type: ReleaseGroupTypeT | {+id: number, +name: string} | null,
     }>,
     +status: ReleaseStatusT | null,
-    +type: ReleaseGroupTypeT | null,
+    +type: ReleaseGroupTypeT | {+id: number, +name: string} | null,
   },
 };
 
 type Props = {
   +edit: EditReleaseAttributesEditT,
 };
 
+function getTypeName(type) {
+  return type
+    // If there's no type.gid then it's a historic "secondary" type
```

Comment:
Since this is itself a historic edit type, does this ever happen here?

Reply:
Yes, for stuff that remained a primary type (Album, Single, EP, Other) this uses $loaded, but for the rest (Compilation, Live, etc) it uses the type mocks.